### PR TITLE
Enhancement (ci): Use `docker build` secrets within context of `docker build`

### PR DESCRIPTION
**Pull request by leojonathanoh**

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -188,6 +187,9 @@ if [ "$PIPELINE" = 'build' ]; then
         time docker pull "$GAME_IMAGE" || true
     fi
     date
+    export DOCKER_BUILDKIT=1
+    export STEAM_USERNAME
+    export STEAM_PASSWORD
```
Comment (joeltimothyoh) [inline build.sh:192]: ok. better to not have duplicate lines though, maybe move them up to after https://github.com/startersclan/docker-sourceservers/pull/131/files#diff-4d2a8eefdf2a9783512a35da4dc7676a66404b6f3826a8af9aad038722da6823R182

**Review by joeltimothyoh — CHANGES_REQUESTED** [review]

File: build.sh (hunk 1 above)
Comment (leojonathanoh) [reply build.sh:192]: For better specificity and security another way is to use:

```sh
DOCKER_BUILDKIT=1 STEAM_USERNAME="$STEAM_USERNAME" STEAM_PASSWORD="$STEAM_PASSWORD" time docker build ...
```

which ensures that no other binary has access to the secrets.

Comment (joeltimothyoh) [reply build.sh:192]: ok got to make sure it works

Comment (leojonathanoh) [reply build.sh]:
```
$ abc1=1 abc2=2 time sh -c 'env|grep abc'
abc1=1
abc2=2
real    0m 0.00s
user    0m 0.00s
sys     0m 0.00s
```

File: build.sh [hunk 2]
```diff
@@ -188,7 +188,7 @@ if [ "$PIPELINE" = 'build' ]; then
         time docker pull "$GAME_IMAGE" || true
     fi
     date
-    time docker build \
+    STEAM_USERNAME="$STEAM_USERNAME" STEAM_PASSWORD="$STEAM_PASSWORD" time docker build \
```
Comment (joeltimothyoh) [inline build.sh:191]: reverted `export DOCKER_BUILDKIT=1` to original position

Comment (leojonathanoh) [reply build.sh]: sorry, i forced push, thought i was making the change.

Comment (leojonathanoh) [reply build.sh]: better for  `DOCKER_BUILDKIT=1` to be on this line, for better specificity that it pertains only to `docker build` than a global `export`.

Comment (leojonathanoh) [reply build.sh]: `hlds` working.

Comment (joeltimothyoh) [reply build.sh]: better to use the global for more consistency in `docker` userd

Comment (leojonathanoh) [reply build.sh]: `DOCKER_BUILDKIT=1` is only relevant to `docker build` and `docker buildx ...` commands

Comment (leojonathanoh) [reply build.sh]: e.g. https://github.com/startersclan/docker-hlstatsxce-daemon/actions/runs/4397801757/jobs/7701187270#step:10:121

Comment (joeltimothyoh) [reply build.sh]: that's for logs where verbosity may be preferred. can't say the same should be used in scripts

**Review by joeltimothyoh — APPROVED** [review]

File: build.sh (hunk 2 above)
Comment (leojonathanoh) [reply build.sh:191]: even though this is a long imperative script, the way it is writen with `\` for each line is a declarative programming approach, so long flags will be more readable.

**Pull request merged**